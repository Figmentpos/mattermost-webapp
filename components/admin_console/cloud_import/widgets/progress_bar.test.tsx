// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {FormattedMessage} from 'react-intl';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import ProgressBar from './progress_bar';

describe('admin_console/progress_bar', () => {
    test('progress bar match snapshot', () => {
        const percent = 10;
        const wrapper = shallow(
            <ProgressBar
                percentage={percent}
                width={50}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('progress bar shows 10%', () => {
        const percent = 10;
        const wrapper = mountWithIntl(
            <ProgressBar
                percentage={percent}
                width={50}
            />,
        );
        expect(wrapper).toMatchSnapshot();
        const span = wrapper.find('span');
        expect(span.last().text()).toBe('10%');
    });

    test('progress bar shows right title', () => {
        const percent = 10;
        const title = (
            <FormattedMessage
                id='admin.general.importInProgress'
                defaultMessage='Import in Progress'
            />
        );
        const wrapper = mountWithIntl(
            <ProgressBar
                percentage={percent}
                width={50}
                title={title}
            />,
        );
        expect(wrapper).toMatchSnapshot();
        const span = wrapper.find('span');
        expect(span.first().text()).toBe('Import in Progress');
    });
});
