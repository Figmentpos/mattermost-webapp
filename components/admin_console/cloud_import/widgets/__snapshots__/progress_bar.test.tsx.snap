// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/progress_bar progress bar match snapshot 1`] = `
<div
  className="ProgressBar  null"
  onClick={[Function]}
>
  <svg
    fill="none"
    height="4"
    viewBox="0 0 50 4"
    width={50}
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="#3D3C40"
      fillOpacity="0.16"
      height="4"
      rx="2"
      width={50}
    />
    <rect
      fill="#166DE0"
      height="4"
      rx="2"
      width="5"
    />
  </svg>
  <span
    className="percetageText"
  >
    10%
  </span>
</div>
`;

exports[`admin_console/progress_bar progress bar shows 10% 1`] = `
<ProgressBar
  percentage={10}
  width={50}
>
  <div
    className="ProgressBar  null"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="4"
      viewBox="0 0 50 4"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#3D3C40"
        fillOpacity="0.16"
        height="4"
        rx="2"
        width={50}
      />
      <rect
        fill="#166DE0"
        height="4"
        rx="2"
        width="5"
      />
    </svg>
    <span
      className="percetageText"
    >
      10%
    </span>
  </div>
</ProgressBar>
`;

exports[`admin_console/progress_bar progress bar shows right title 1`] = `
<ProgressBar
  percentage={10}
  title={
    <FormattedMessage
      defaultMessage="Import in Progress"
      id="admin.general.importInProgress"
    />
  }
  width={50}
>
  <div
    className="ProgressBar  null"
    onClick={[Function]}
  >
    <FormattedMessage
      defaultMessage="Import in Progress"
      id="admin.general.importInProgress"
    >
      <span>
        Import in Progress
      </span>
    </FormattedMessage>
    <svg
      fill="none"
      height="4"
      viewBox="0 0 50 4"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#3D3C40"
        fillOpacity="0.16"
        height="4"
        rx="2"
        width={50}
      />
      <rect
        fill="#166DE0"
        height="4"
        rx="2"
        width="5"
      />
    </svg>
    <span
      className="percetageText"
    >
      10%
    </span>
  </div>
</ProgressBar>
`;
